genesyscloud/resource_genesyscloud_architect_user_prompt.go

```diff
@@ -0,0 +1,1081 @@
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectAPI := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Reading User Prompt %s", d.Id())
+	userPrompt, resp, getErr := architectAPI.GetArchitectPrompt(d.Id())
+	if getErr != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Failed to read User Prompt %s: %s", d.Id(), getErr)
+	}
+
+	d.Set("name", *userPrompt.Name)
+	d.Set("description", *userPrompt.Description)
+
+	if userPrompt.Name != nil {
+		d.Set("name", *userPrompt.Name)
+	} else {
+		d.Set("name", nil)
+	}
+
+	if userPrompt.Description != nil {
+		d.Set("description", *userPrompt.Description)
+	} else {
+		d.Set("description", nil)
+	}
+
+	d.Set("resources", flattenPromptResources(userPrompt.Resources))
+
+	log.Printf("Read Audio Prompt %s %s", d.Id(), *userPrompt.Id)
+	return nil
+}
+
+func updateUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Updating user prompt %s", name)
+	_, _, err := architectApi.PutArchitectPrompt(d.Id(), platformclientv2.Prompt{
+		Name:        &name,
+		Description: &description,
+	})
+	if err != nil {
+		return diag.Errorf("Failed to update user prompt %s: %s", name, err)
+	}
+
+	diagErr := updatePromptResource(d, architectApi, sdkConfig)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Printf("Updated User Prompt %s", d.Id())
+	return readUserPrompt(ctx, d, meta)
+}
+
+func deleteUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Deleting user prompt %s", name)
+	if _, err := architectApi.DeleteArchitectPrompt(d.Id(), true); err != nil {
+		return diag.Errorf("Failed to delete user prompt %s: %s", name, err)
+	}
+	log.Printf("Deleted user prompt %s", name)
+
+	return withRetries(ctx, 30*time.Second, func() *resource.RetryError {
+		_, resp, err := architectApi.GetArchitectPrompt(d.Id())
+		if err != nil {
+			if resp != nil && resp.StatusCode == 404 {
+				// User prompt deleted
+				log.Printf("Deleted user prompt %s", name)
+				return nil
+			}
+			return resource.NonRetryableError(fmt.Errorf("Error deleting user prompt %s: %s", name, err))
+		}
+		return resource.RetryableError(fmt.Errorf("User prompt %s still exists", name))
+	})
+}
+
+func uploadPrompt(uploadUri *string, filename *string, sdkConfig *platformclientv2.Configuration) (string, []byte) {
+	accessToken := sdkConfig.AccessToken
+
+	client := &http.Client{}
+	file, err := os.Open(*filename)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
It's probably better practice to have this function return the error and the caller handle it by returning it in the diag.Errorf function